src/connectors/jamf_inventory.py

```diff
@@ -75,7 +82,39 @@ async def main(table_name):
 
 def ingest(table_name, options):
     global HEADERS
-    creds = options.get('credentials', '')
-    HEADERS = {'Authorization': f'Basic {creds}', 'Accept': 'application/json'}
+    token = getAccessToken(options=options['credentials'])
+    HEADERS = {'Authorization': f'Bearer {token}', 'Accept': 'application/json'}
     return asyncio.get_event_loop().run_until_complete(main(f'data.{table_name}'))
 
+
+# options is a dict containing cliendId and clientSecret
+def getAccessToken(credentials: str) -> str:
+
+    # credentials stored in format {"client_id":"some-id","client_secret":"some-secret"}
+    client_cred = json.loads(credentials)
+
+    jamf_token_api_path = "https://snowflake.jamfcloud.com/api/oauth/token"
+
+    data = {
+        'client_id': client_cred[CLIENT_ID_KEY],
+        'grant_type': 'client_credentials',
+        'client_secret': client_cred[CLIENT_SECRET_KEY],
+    }
+
+    response = requests.post(
+        jamf_token_api_path,
+        data=urlencode(data),
```

Comment:
per [the docs](https://requests.readthedocs.io/en/latest/user/quickstart/#more-complicated-post-requests), you can pass a dict here and then you don't need to set headers or call urlencode.